node/pkg/nodeinit/calico-init_linux.go

```diff
@@ -65,11 +76,55 @@ func ensureBPFFilesystem() error {
 	}
 
 	// If we get here, the BPF filesystem is not mounted.  Try to mount it.
-	logrus.Info("BPF filesystem is not mounted.  Trying to mount it...")
-	err = syscall.Mount(bpfFSMountPoint, bpfFSMountPoint, "bpf", 0, "")
+	logrus.Info("BPF filesystem is not mounted. Trying to mount it...")
+	err = syscall.Mount(bpf.DefaultBPFfsPath, bpf.DefaultBPFfsPath, "bpf", 0, "")
 	if err != nil {
 		return fmt.Errorf("failed to mount BPF filesystem: %w", err)
 	}
 	logrus.Info("Mounted BPF filesystem.")
 	return nil
 }
+
+// ensureCgroupV2Filesystem() enters the cgroup and mount namespace of the process
+// with PID 1 running on a host to allow felix running in calico-node to access the root of cgroup namespace.
+// This is needed by felix to attach CTLB programs and implement k8s services correctly.
+func ensureCgroupV2Filesystem() error {
+	cgroupRootPath := "/initproc"
+
+	// Check if the Cgroup2 filesystem is mounted at the expected location.
+	logrus.Info("Checking if Cgroup2 filesystem is mounted.")
+	mountInfoFile := path.Join(cgroupRootPath, "mountinfo")
+	mounts, err := os.Open(mountInfoFile)
+	if err != nil {
+		return fmt.Errorf("failed to open %s: %w", mountInfoFile, err)
+	}
+	scanner := bufio.NewScanner(mounts)
+	for scanner.Scan() {
+		// An example line in mountinfo file:
+		// 35 24 0:30 / /sys/fs/cgroup rw,nosuid,nodev,noexec,relatime shared:9 - cgroup2 cgroup2 rw,nsdelegate,memory_recursiveprot
+		line := scanner.Text()
+		mountPoint := strings.Split(line, " ")[4]                      // 4 is the index to mount points in mountinfo files
+		fsType := strings.Split(strings.Split(line, " - ")[1], " ")[0] // fsType is the first string after -
```

Comment:
I don't think you can assume that that column in the output will always be `-` and your code will panic if there's no `-` in the output.  What happens if there is a space in the mount point, did you figure that out?

Reply:
I tested a mount point with space, and the space character was translated into `\040` which is it's octal ascii code. 

`164 31 0:30 / /home/mazdak/src/calico/node/some\040-\040path rw,relatime shared:61 - cgroup2 none rw`

Reply:
Ah, I misread it as saying the optional params were after the `-`.  Good catch!